compute_tools/src/extensions.rs

```diff
@@ -0,0 +1,103 @@
+use compute_api::responses::Extension;
+use compute_api::responses::ExtenstionsList;
+use url::Url;
+use std::collections::HashMap;
+
+// TODO limit the number of databases we query
+// TODO write proper version comparison
+//
+// TODO call it from somewhere, i.e. from ControlPlane after StartCompute (?)
```

Comment:
>put the result into the shared compute state, Then in the API we will just read the state instead of getting the data right in the handler.

+1

>Right now, I think it's fine to gather data only at start, but we can also add a background thread that will re-gather the data

I was going to add CreateExtension DDL handling to [forward_ddl](https://github.com/neondatabase/neon/blob/main/pgxn/neon/control_plane_connector.c#L771) hook and push these updates from postgres to [cplane](https://github.com/neondatabase/cloud/blob/57bccf2deb51405f60e085befae9d053c2b13add/goapp/controlplane/internal/app/computeapiv1/internal/controller/computes.go#L238).
Same way as we handle updates of roles and databases.
Extension updates don't necessarily need to be real-time now, but I don't see any reason, why communication through compute would be better.
WDYT?

Reply:
> CreateExtension DDL

+ alter extension, I think

Yeah, this makes sense, also thought about it. BTW, are extension statements (CREATE, ALTER) transactional? If yes, then it's probably better to put handling into the commit hook? With DBs it's just not transactional, so that's why this location, iirc

I still wonder what we should do with drop extension. If the user drops it in one DB, it doesn't necessarily mean there are no other DBs still having this extension. If we do not handle that, list of extensions in cplane will only grow in time

Reply:
> BTW, are extension statements (CREATE, ALTER) transactional? 

yes, they are

>If yes, then it's probably better to put handling into the commit hook? With DBs it's just not transactional, so that's why this location, iirc

`control_plane_connector` already sends all changes on commit (it also handles roles, which are transactional).
So, IIUC, no changes are needed there.

Reply:
will do in a follow-up PR